src/charm.py

```diff
@@ -128,9 +128,9 @@ def __init__(self, *args):
                 config=to_tuple(ActiveStatus()),
                 rules=to_tuple(ActiveStatus()),
                 retention=to_tuple(ActiveStatus()),
-            )
+            ),
+            fresh_install=True,
```

Comment:
it's not clear to me what this does, and why this change was necessary. I realize this is hard to test as we'd need an integration test to redeploy the charm, or simulate pod churn? 
But at least let's add a detailed comment to explain what's going on.

Reply:
if you do think this is testable though, please let's write a test

Reply:
It was introduced in #393 and there's an itest.

Reply:
Yeah this code is from a previous commit, but like @PietroPasotti I found it tough to understand the importance at first.  Maybe a comment or link to an issue explaining it could help.  Although it does get better explanation in the relevant part of `configure` further down

Reply:
Yep, @IbraAoad documented this in two places next to where it's used:
- https://github.com/canonical/loki-k8s-operator/blob/fd3e58407ef55c5150ac5b54bf42dd4fb8a7dadc/src/charm.py#L492
- https://github.com/canonical/loki-k8s-operator/blob/fd3e58407ef55c5150ac5b54bf42dd4fb8a7dadc/src/charm.py#L252